provider/azure/environ.go

```diff
@@ -842,7 +873,30 @@ func (env *azureEnviron) createVirtualMachine(
 			AvailabilitySet: availabilitySetSubResource,
 		},
 		DependsOn: vmDependsOn,
-	})
+	}
+	// For controllers, check to see if we need to assign a managed identity resource to the vm.
+	if instanceConfig.IsController() {
+		var managedIdentity string
+		if env.cloud.Credential.AuthType() == cloud.InstanceRoleAuthType {
+			// Add a new controller after bootstrap (enable-ha).
+			managedIdentity = env.cloud.Credential.Attributes()[credManagedIdentity]
+		} else if instanceConfig.Bootstrap.BootstrapMachineConstraints.HasInstanceRole() {
```

Comment:
This causes a panic when enabling HA. `IsController` is true, but we are not bootstrapping.